library/src/main/java/com/novoda/downloadmanager/DownloadManager.java

```diff
@@ -159,4 +145,32 @@ private void executeGetAllDownloadBatchStatuses(AllBatchStatusesCallback callbac
 
         callback.onReceived(downloadBatchStatuses);
     }
+
+    @Override
+    public void getFirstLocalPathForDownloadWithMatching(final String networkUri, final DownloadFilePathCallback callback) {
+        if (downloadService == null) {
+            executor.submit(
```

Comment:
Extracted a `Runnable` that knows to wait for a `lock` before performing a given action.